demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c

```diff
@@ -1028,10 +1028,6 @@ CK_RV xProvisionDevice( CK_SESSION_HANDLE xSession,
                 configPRINTF( ( "Warning: could not clean-up old crypto objects. %d \r\n", xResult ) );
             }
         }
-        else
```

Comment:
The change that's being reverted here was an attempt to fix another bug, correct? Do you have an alternative fix for that other bug?

Reply:
In hindsight this was not the right thing to do, I fixed the bug accidentally, but this code was correct. The real issue was actually the call to generatekey, which tried to generate an ECDSA key on ST and TI, causing a hard fault.